lib/Frontend/LoadedModuleTrace.cpp

```diff
@@ -0,0 +1,47 @@
+//===--- LoadedModuleTrace.cpp - Manipulating a loaded module trace file --===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+#include "swift/Frontend/LoadedModuleTrace.h"
+
+#include "swift/Basic/JSONSerialization.h"
+#include "llvm/Support/raw_ostream.h"
+
+using namespace swift::loaded_module_trace;
+using namespace swift;
+
+namespace swift {
+namespace json {
+template <typename T> struct ArrayTraits<std::vector<T>> {
```

Comment:
This seems like it should just be sunk into JSONSerialization. Otherwise someone will end up adding their own.